lib/fizzy/execute.cpp

```diff
@@ -324,6 +326,19 @@ inline DstT extend(SrcT in) noexcept
         return DstT{in};
 }
 
+template <typename SrcT, typename DstT>
+inline bool trunc(OperandStack& stack) noexcept
+{
+    using boundaries = trunc_boundaries<SrcT, DstT>;
+
+    const auto input = stack.top().as<SrcT>();
+    if (std::isnan(input) || input <= boundaries::lower || input >= boundaries::upper)
```

Comment:
Why doesn't this need to check for `inf`/`-inf`?

Reply:
Because always: `-inf <= boundaries::lower` and `inf >= boundaries::upper`.

Reply:
Oh, actually, this can be optimized :/

Reply:
Even if that is true (seemingly it is because the tests pass), we should have a static_assert here for that.

Reply:
I'm not sure what assert you would like to have here.

Reply:
```
-inf <= boundaries::lower
inf >= boundaries::lower
```

Reply:
No, this does not make sense to me. Every value is `>= -inf`.